src/data/fetchStats.ts

```diff
@@ -0,0 +1,29 @@
+import { writeFile } from 'fs/promises';
+
+fetchStatsOrDoNothing();
+
+export async function fetchStatsOrDoNothing() {
+  try {
+    const stats = await fetchStats();
+    await storeStats(stats);
+  } catch {}
+}
+
+async function storeStats(stats: Awaited<ReturnType<typeof fetchStats>>) {
+  return writeFile('./src/data/stats.json', prettyJSONStringify(stats) + "\n");
```

Comment:
Was there a strong reason you had to persist the stats to a file? If not, I'm tempted to just query the API every time. It would only need to happen at site build time, anyway

Reply:
Just wanted to have a fallback if the API fetching fails (downtime / regional blocks / whatever). To e.g. be able to build without Internet connection, but also without statically hardcoding any data.

Does removing the persistence  logic have any benefits?

Reply:
Mainly that the data file would constantly be changing, resulting in a ton of Git commit noise. If we keep it, it would need to be added to .gitignore and regenerated locally for each new repo clone. Since the deployments are done on GitHub Actions, the data would need to be pulled in fresh each time for that, anyway.

I understand the potential benefit for building the site when the API is unreachable, though. I'll go ahead and gitignore it for the time being. Thanks for thinking of this!